pkg/storage/tsdb/config.go

```diff
@@ -114,6 +115,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxTSDBOpeningConcurrencyOnStartup, "experimental.tsdb.max-tsdb-opening-concurrency-on-startup", 10, "limit the number of concurrently opening TSDB's on startup")
 	f.DurationVar(&cfg.HeadCompactionInterval, "experimental.tsdb.head-compaction-interval", 1*time.Minute, "How frequently does Cortex try to compact TSDB head. Block is only created if data covers smallest block range. Must be greater than 0 and max 5 minutes.")
 	f.IntVar(&cfg.HeadCompactionConcurrency, "experimental.tsdb.head-compaction-concurrency", 5, "Maximum number of tenants concurrently compacting TSDB head into a new block")
+	f.IntVar(&cfg.StripeSize, "experimental.tsdb.stripe-size", 14, "Power of 2 to use for the tsdb stripe size. Reducing this will decrease memory footprint, but can negatively impact performance.")
```

Comment:
You mention it's a power of 2 and the default is 14, but then when you configure the `tsdb.Options` you're not actually applying the "power of 2" so the effective size is 14 and not 2^14.

Reply:
Oh, I believe you're right. I have assumed it will be used as a power of 2 internally, but we actually need to do it ourselves (```StripeSize:        1 << i.cfg.TSDBConfig.StripeSize,```).

Reply:
Lol you're right. I shouldn't have tried to make this PR, too tired :joy: 

Added the shift, into the open options.